### PR TITLE
Centralized logging with ELK

### PR DESCRIPTION
**Pull request by PierreBesson**
Provide:
- a docker compose file to easily setup Elasticsearch, Logback and Kibana.
- a logstash configuration file that retrieve logs through the syslog protocol on UDP port 5000

Modify:
- Logback configurations for the registry, gateway and app1 to forward logs with the syslog protocol.

What can be improved:
- Better parsing of Spring boot log format (so that it can detect stacktraces). Since I am no longer reading the logs though a file, the formating is different and I cannot use [this article's configuration](http://knes1.github.io/blog/2015/2015-08-16-manage-spring-boot-logs-with-elasticsearch-kibana-and-logstash.html) as is.
- Adding more informations to the logs so we can know which app produced which logs.

### CODE REVIEW

File: elk/elk.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  image: elasticsearch:1.7.1
+  ports:
+    - "9200:9200"
+    - "9300:9300"
+elk-logstash:
+  container_name: elk-logstash
+  image: logstash:1.5.3
+  volumes:
+    - ./logstash/:/config-dir
+    - /tmp/:/log-tmp/
```
Comment (pascalgrimaud) [inline elk/elk.yml]: no need this line `/tmp/:/log-tmp/` because you didn't use log
But keep it if you plan to use log later

Comment (PierreBesson) [inline elk/elk.yml]: Thank you @pascalgrimaud and big ups for writing the original elk.yml !

**Comment by PierreBesson** [comment]
Even though I know that we have switched to work on the main generator directly. I commit this here as a reference.
I think this should be the final log config. Having logback produce JSON by itself is by far the best way to integrate with ELK much better than parsing another format with logstash filter.

**Comment by jdubois** [comment]
Yes it's great.
I'm keeping this repo alive until this is integrated in the main generator.

**Pull request closed without merging**